scripts/run_performance_job.py

```diff
@@ -336,72 +683,94 @@ def run_performance_job(args: RunPerformanceJobArgs):
         
         # build MemoryConsumption
         RunCommand([
-            dotnet_path, "publish", 
+            dotnet_executable_path, "publish", 
             "-c", "Release", 
-            "-o", os.path.join(performance_setup_data.payload_directory, "MemoryConsumption"),
+            "-o", os.path.join(payload_dir, "MemoryConsumption"),
             "-f", framework,
             "-r", runtime_id,
             "--self-contained",
             os.path.join(args.performance_repo_dir, "src", "tools", "ScenarioMeasurement", "MemoryConsumption", "MemoryConsumption.csproj"),
             "-p:DisableTransitiveFrameworkReferenceDownloads=true"]).run()
         
+        # build PerfLabGenericEventSourceForwarder
+        RunCommand([
+            dotnet_executable_path, "publish", 
+            "-c", "Release", 
+            "-o", os.path.join(payload_dir, "PerfLabGenericEventSourceForwarder"),
+            "-f", framework,
+            "-r", runtime_id,
+            os.path.join(args.performance_repo_dir, "src", "tools", "PerfLabGenericEventSourceForwarder", "PerfLabGenericEventSourceForwarder", "PerfLabGenericEventSourceForwarder.csproj"),
+            "-p:DisableTransitiveFrameworkReferenceDownloads=true"]).run()
+        
+        # build PerfLabGenericEventSourceLTTngProvider
+        if args.os_group != "windows" and args.os_group != "osx" and args.os_version == "2204":
```

Comment:
Is this check correct? Because even in the latest run I don't see `PerfLabGenericEventSourceLTTngProvider` compiled.

Reply:
I believe so, this is what the current yaml looks like that calls this: https://github.com/dotnet/performance/blob/1d11344caab47e35dcf38d5f17dacb924b35fcba/eng/performance/scenarios.yml#L258

This line has two conditions on it in the yaml:
`- ${{ if and(ne(parameters.osName, 'windows'), ne(parameters.osName, 'osx')) }}:`
`condition: and(succeeded(), contains(variables['System.JobDisplayName'], '2204'))`

So this is the combination of those two. Is there a specific pipeline run you were expecting to see it in that it isn't being run in?

Reply:
Ah, my bad, there is an error there. The check for os_version is not correct.

Reply:
Is the `os_version` really `2204` and not like `22.04`?

The most recent run here is still failing on _ubuntu 2204 x64 scenarios_ and I don't see `PerfLabGenericEventSourceLTTngProvider` compilation in output...